packages/docs-ui/src/components/LinkableHeading/LinkableHeading.tsx

```diff
@@ -0,0 +1,48 @@
+import React, { type ReactNode, type ComponentProps } from 'react';
+import { Heading, Box, IconLink, Link } from 'braid-design-system';
+import * as styles from './LinkableHeading.css';
+
+type HeadingProps = ComponentProps<typeof Heading>;
+
+type LinkableHeadingProps = {
+  level?: HeadingProps['level'];
+  component?: HeadingProps['component'];
+} & ({ children: string } | { children: ReactNode; label: string });
+
+const slugify = (string: string) =>
+  string
+    .replace(/[\s?]/g, '-')
+    .replace('--', '-')
+    .replace(/-$/, '')
+    .toLowerCase();
+
+export const LinkableHeading = ({
+  level = '3',
+  component,
+  ...restProps
+}: LinkableHeadingProps) => {
+  const label = 'label' in restProps ? restProps.label : restProps.children;
+  const slug = slugify(label);
+
+  return (
+    <Box display="flex">
+      <Box className={styles.linkableHeading} display="inlineBlock">
+        <Box id={slug} position="absolute" />
+        <Link href={`#${slug}`}>
+          <Heading level={level} component={component}>
+            {restProps.children}{' '}
+            <Box
+              component="span"
+              transition="fast"
+              marginLeft="xxsmall"
```

Comment:
Do we need the space _and_ the margin?

Reply:
Hmm not sure, this is just a direct copy of the original component. What's giving space other than the margin?

Reply:
There a space character on line 33.

Reply:
Oo yep nice catch! Have made a fix in the follow up PR.